android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -90,14 +89,11 @@ public object WorkerBinder {
   @JvmStatic
   @VisibleForTesting
   public fun bind(mappedLifecycle: Observable<WorkerEvent>, worker: Worker): WorkerUnbinder {
-    val unbindFlow = MutableSharedFlow<WorkerEvent>(0, 1, BufferOverflow.DROP_OLDEST)
-
-    val workerLifecycle = merge(mappedLifecycle.asFlow(), unbindFlow)
-      .transformWhile { emit(it) ; it != WorkerEvent.STOP }
-      .asObservable()
-
-    bindToWorkerLifecycle(workerLifecycle, worker)
-    return WorkerUnbinder { unbindFlow.tryEmit(WorkerEvent.STOP) }
+    val disposable = mappedLifecycle
```

Comment:
I noticed there is also on line 153. Should this be updated accordingly as well? The params list seems very similar. It just differs on returning type 

@JvmStatic
  public fun bindToWorkerLifecycle(
    workerLifecycle: Observable<WorkerEvent>,
    worker: Worker
  ) {
    workerLifecycle.subscribe { workerEvent: WorkerEvent ->
      when (workerEvent) {
        WorkerEvent.START -> worker.onStart(WorkerScopeProvider(workerLifecycle.hide()))
        else -> worker.onStop()
      }
    }
  }

Reply:
Great points. In this case maybe this could be a good moment to add the deprecation notice there as well

Reply:
Deprecated it with:
```kotlin
  @Deprecated(
    message = """
      This method is unsafe because it assumes that the 'workerLifecycle' Observable completes after it
      emits 'WorkerEvent.STOP'. If it does not complete, subscription will be leaked.
    """,
    replaceWith = ReplaceWith("bind(workerLifecycle, worker)")
  )
```